java-lib/src/main/java/com/wavefront/data/Validation.java

```diff
@@ -203,9 +209,15 @@ public static void validateSpan(Span span, @Nullable ValidationConfiguration con
           throw new IllegalArgumentException("WF-416: Point tag key has illegal character(s): " + tagK);
         }
         if (tagV.length() > config.getSpanAnnotationsValueLengthLimit()) {
-          ERROR_COUNTERS.get("spanAnnotationValueTooLong").inc();
-          throw new IllegalArgumentException("WF-433: Span annotation value is too long (" + tagV.length() +
-              " characters, max: " + config.getAnnotationsValueLengthLimit() + "): " + tagV);
+          if (blockedLoggingRateLimiter.tryAcquire()){
+            log.warning("[" + span.getCustomer() + "] Span trimmed due to tagV length: " +
+                annotation.getKey() + " limit for: " + span.getCustomer() + " is: " +
+                config.getAnnotationsValueLengthLimit() + ", found: " + annotation.getValue().length()
+                + ", span: " + span);
+            // trim the tag value to the allowed limit
+            annotation.setValue(annotation.getValue().substring(0, config.getSpanAnnotationsValueLengthLimit()));
```

Comment:
Critical: this should not be inside `if(tryAcquire)` block, we want to fix every span, not just one per second